spanner/api/Spanner.Samples/CustomTimeoutsAndRetriesAsync.cs

```diff
@@ -0,0 +1,66 @@
+// Copyright 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START spanner_set_custom_timeout_and_retry]
+
+using Google.Api.Gax;
+using Google.Api.Gax.Grpc;
+using Google.Cloud.Spanner.Common.V1;
+using Google.Cloud.Spanner.Data;
+using Google.Cloud.Spanner.V1;
+using static Google.Cloud.Spanner.V1.TransactionOptions.Types;
+using Grpc.Core;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+public class CustomTimeoutsAndRetriesAsyncSample
+{
+    public async Task<long> CustomTimeoutsAndRetriesAsync(string projectId, string instanceId, string databaseId)
+    {
+        // Create a SessionPool.
+        SpannerClient client = SpannerClient.Create();
+        SessionPool sessionPool = new SessionPool(client, new SessionPoolOptions());
+
+        // Acquire a session with a read-write transaction to run a query.
+        DatabaseName databaseName =
+            DatabaseName.FromProjectInstanceDatabase(projectId, instanceId, databaseId);
+        TransactionOptions transactionOptions = new TransactionOptions
+        {
+            ReadWrite = new ReadWrite()
+        };
+        PooledSession session = await sessionPool.AcquireSessionAsync(
+            databaseName, transactionOptions, CancellationToken.None);
+
+        ExecuteSqlRequest request = new ExecuteSqlRequest
+        {
+            Sql = "INSERT Singers (SingerId, FirstName, LastName) VALUES (20, 'George', 'Washington')"
```

Comment:
I see what you mean. I agree that it would be nice to keep the syntax consistent across all samples in all languages.

In this case though, the most important thing is consistency of the same sample across all the languages as they will appear side-by-side in the documentation. For the custom timeout and retry setting samples we have so far [Java](https://github.com/GoogleCloudPlatform/java-docs-samples/pull/3445/files) and [Go](https://github.com/GoogleCloudPlatform/golang-samples/pull/1594/files) use `INSERT` and [Node.js](https://github.com/googleapis/nodejs-spanner/blob/f2f8827c58d1c22ac031a091b8c36ad06d64d42b/samples/dml.js#L683) uses the `insert` API so just want to make sure that we're consistent with those.